### PR TITLE
Update Circle and add Travis to generate documentation

### PR DESCRIPTION
**Pull request by katydecorah**
In https://github.com/mapbox/documentation/issues/527, the docs team proposed that generated documentation could live in the `publisher-production` branches of the source repo.

This PR prepares Circle to handle this change, including:
- Add a new step in Circle to "Trigger documentation deploy steps"
  - Upon a new release, this step will trigger Travis to run in this repo.
  - Travis will then build the documentation and run a script to generate a PR against `publisher-production` with the generated documentation for the latest release.
- The original Circle job to "Trigger external deploy steps" will remain. This step will bump the constants and other metadata-related files in https://github.com/mapbox/ios-sdk.

## Notes

The `publisher-production` (and `publisher-staging`) branches will only contain the generated documentation and any files required to publish them: https://github.com/mapbox/mapbox-gl-native-ios/tree/publisher-staging

## To do
- [x] Enable Travis in this repo.
- [x] Add GitHub token to Travis settings.
- [x] Update ios-sdk Travis job to remove steps to build documentation and only update metadata files.
- [x] Determine: Could `docs-github-pull-request.js` be a package? Since we'll likely be using this in several repos, it would help reduce footprint in each repo.
- [x] Add @MapboxCI as contributor with write access to this repo.
- [ ] (?) Add script to push previous version to the S3 archive site.
- [x] Test the pipeline.
- [x] Add inline documentation to configuration files

## Publisher steps

- [x] Enable publisher on this repository
- [x] Register the repository under docs subdomain.
- [x] Create publisher-staging branch
- [x] Create publisher-production branch

## How to test

1. In Travis, click More Options > Trigger build.
2. Branch: `docs`
3. Custom commit mesage: `[maps] docs pipeline test`
4. Custom config:
```yml
merge_mode: deep_merge
env:
 - SDK_FLAVOR=maps RELEASE_TAG=6.1.0
```

---

cc @danswick

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  include:
+    - name: "Build Maps Release Docs"
+      os: osx
+      osx_image: xcode10.2
+      if: env(SDK_FLAVOR) = maps AND env(RELEASE_TAG) IS PRESENT
+      script:
+        - npm link
+        - git config user.email "MapboxCI@users.noreply.github.com"
+        - ./scripts/docs-update-maps.sh $RELEASE_TAG
+        - docs-release-pull-request -s $SDK_FLAVOR -t $RELEASE_TAG -b publisher-staging -p ios -r mapbox-gl-native-ios
```
Comment (katydecorah) [inline .travis.yml:19]: Before merge, we can remove the branch flag (default is publisher-production)

```suggestion
        - docs-release-pull-request -s $SDK_FLAVOR -t $RELEASE_TAG -p ios -r mapbox-gl-native-ios
```

**Comment by katydecorah** [comment]
I'm still stumped. Travis continues to error during `git push`:

> remote: Invalid username or password.

Since this script works in ios-sdk, I think it's related to permissions. I think we should double check that the `GITHUB_TOKEN` in the Travis configuration is correct. (@danswick can you help me with this one.)

**Comment by katydecorah** [comment]
Update the GITHUB_TOKEN fixed the issue. We got a build in #435 

To do
- [ ] Fix label, it is currently "undefined".
- [ ] Only commit the generated api docs folder.

File: scripts/docs-update-maps.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+mv -v $OUTPUT/* "./$VERSION"
+
+# Do we generate docs for pre-releases?
+#if [[ $( echo ${VERSION} | awk '/[0-9]-/' ) ]]; then
+#    step "Skipping website updates because ${VERSION} is a pre-release"
+#    exit 0
+#fi
+
+step "Switching branch to publisher-production"
+git checkout publisher-staging
```
Comment (katydecorah) [inline scripts/docs-update-maps.sh:39]:
```suggestion
git checkout publisher-production
```

To do before merge!

**Comment by katydecorah** [comment]
I think I need to change the git depth for the shell scripts, it can't find `publisher-staging`

**Comment by katydecorah** [comment]
The build at https://github.com/mapbox/mapbox-gl-native-ios/pull/446 worked great. I just added one more commit that adds `[ci skip]` to the commit message - this will prevent circle from running for these generated docs commits https://circleci.com/docs/2.0/skip-build/

File: scripts/docs-update-maps.sh [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+mv -v $OUTPUT/* "./$VERSION"
+
+# Do we generate docs for pre-releases?
+#if [[ $( echo ${VERSION} | awk '/[0-9]-/' ) ]]; then
+#    step "Skipping website updates because ${VERSION} is a pre-release"
+#    exit 0
+#fi
+
+step "Switching branch to publisher-production"
+git checkout origin/publisher-staging
```
Comment (katydecorah) [inline scripts/docs-update-maps.sh:38]: Update this before merge:

```suggestion
git checkout origin/publisher-production
```

**Comment by katydecorah** [comment]
@danswick I think this is ready for review!

Travis just completed a test PR at: https://github.com/mapbox/mapbox-gl-native-ios/pull/447

**Review by danswick — APPROVED** [review]
This looks great to me, @katydecorah 👍

File: scripts/docs-update-maps.sh [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+function finish { >&2 echo -en "\033[0m"; }
+trap finish EXIT
+
+OUTPUT="/tmp/`uuidgen`"
+
+step "Updating mapbox-gl-native-ios repository…"
+git fetch --all
+git fetch --tags
+
+VERSION=${1}
```
Comment (jmkiley) [inline scripts/docs-update-maps.sh:17]: Because the release tag is formatted as `ios-vX.X.X`, we should probably remove the `ios-v` at this point.

**Review by jmkiley — CHANGES_REQUESTED** [review]

File: scripts/docs-update-maps.sh (hunk 4 above)
Comment (katydecorah) [inline scripts/docs-update-maps.sh:17]: Should do it!

```suggestion
VERSION=${1//ios-v/''}
```

Comment (danswick) [reply scripts/docs-update-maps.sh:17]: Heh, was just about to type this!

**Review by jmkiley — APPROVED** [review]
LGTM!

File: circle.yml [hunk 5]
```diff
@@ -478,7 +478,13 @@ jobs:
       - install-dependencies
       - install-ios-packaging-dependencies
       - run:
-          name: Trigger external deploy steps
+          name: Trigger documentation deploy steps
+          command: |
+            export VERSION_TAG=${CIRCLE_TAG}
+            export DOCS_REPO="mapbox-gl-native-ios"
```
Comment (jmkiley) [inline circle.yml:484]: I think this should be `ios-sdk`. I tried this locally and accidentally created a docs PR in this repo.

**Pull request merged**